source/distributed/tria.cc

```diff
@@ -4556,7 +4431,7 @@ namespace parallel
                   ExcMessage ("Parallel distributed triangulations can only "
                               "be copied, if no refinement is in progress!"));
 
-          mpi_communicator = Utilities::MPI::duplicate_communicator (old_tria_x->get_communicator ());
+          this->mpi_communicator = Utilities::MPI::duplicate_communicator (old_tria_x->get_communicator ());
```

Comment:
This makes it difficult to establish who the owner of the `mpi_communicator` object is and who is responsible for deleting it again. If the base class owns the object, then it needs to be in charge of duplicating it (and later deleting the duplicate).